airflow/providers/weaviate/operators/weaviate.py

```diff
@@ -0,0 +1,87 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from functools import cached_property
+from typing import TYPE_CHECKING, Any, Callable, Collection, Mapping, Sequence
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.weaviate.hooks.weaviate import WeaviateHook
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class WeaviateIngestOperator(BaseOperator):
+    """
+    Operator that store vector in the Weaviate class.
+
+    Operator that accepts input json to generate embeddings on or accepting provided custom vectors
+     and store them in the Weaviate class.
+
+    :param conn_id: The Weaviate connection.
+    :param class: The Weaviate class to be used for storing the data objects into.
+    :param input_json: The JSON representing Weaviate data objects to generate embeddings on (or provides
+    custom vectors) and store them in the Weaviate class. Either input_json or input_callable should
+     be provided.
+    :param input_callable: The callable that provides the input json to generate embeddings on
+        (or provides custom vectors) and store them in the Weaviate class. Either input_text or
+         input_callable should be provided.
+    """
+
+    template_fields: Sequence[str] = ("input_json",)
+
+    def __init__(
+        self,
+        conn_id: str,
+        class_name: str,
+        input_json: dict[str, Any] | None = None,
+        input_callable: Callable[[Any], Any] | None = None,
+        input_callable_args: Collection[Any] | None = None,
+        input_callable_kwargs: Mapping[str, Any] | None = None,
```

Comment:
Still the same point about the "PythonOperator" inside of this one

Reply:
@Taragolis Updated the interface, with your suggestion. :)